docs/static/settings/ecs-compatibility.asciidoc

```diff
@@ -0,0 +1,49 @@
+[[ecs-compatibility]]
+=== ECS Compatibility
+
+Many plugins implement an ECS-compatibility mode, which causes them to produce and manipulate events in a manner that is compatible with the Elastic Common Schema (ECS).
+
+These plugins can also be configured to use their legacy non-ECS behaviour: individually, per-pipeline, or globally.
+Any plugin that supports this mode will also have an `ecs_compatibility` option, which allows you to configure which mode the individual plugin instance should operate in.
+If left unspecified for an individual plugin, the pipeline's `pipeline.ecs_compatibility` setting will be observed.
+
+==== Opting Out
+
+In Logstash 8, these plugins are run in ECS mode by default, but you can opt-out and use the legacy behaviour in one of several ways.
+This can be helpful if you have very complex pipelines that were defined pre-ECS, to allow you to either upgrade them or to avoid doing so independently of your {ls} 8.x upgrade.
+
+===== Specific Plugin Instance
+
+Set `ecs_compatibility => disabled` to override the default value on the plugin instance.
+For example, if you want an instance of the GeoIP Filter to behave as it did before ECS was on by default, you can adjust its definition in your pipeline without affecting any other plugin instances.
+
+[source,text]
+filter {
+  geoip {
+    source => "[host][ip]"
+    ecs_compatibility => disabled
+  }
+}
+
+===== All Plugins in a given Pipeline
+
+If you wish to provide a specific default value for `ecs_compatibility` to _all_ plugins in a pipeline, you can do so with the `pipeline.ecs_compatibility` setting in your pipeline definition in `config/pipelines.yml` or Central Management.
+This setting will be used unless overridden by a specific plugin instance. If unspecified for an individual pipeline, the global value will be used.
+
+[source,yaml]
+- pipeline.id: my-legacy-pipeline
+  path.config: "/etc/path/to/legacy-pipeline.config"
+  pipeline.ecs_compatibility: disabled
+- pipeline.id: my-ecs-pipeline
+  path.config: "/etc/path/to/ecs-pipeline.config"
+  pipeline.ecs_compatibility: v1
+
+NOTE: Until the General Availability of Logstash 8.0.0, any value for `pipeline.ecs_compatibility` other than `disabled` -- including the default value on this pre-release branch -- may have undesireable consequences when performing upgrades.
```

Comment:
This caveat will go away before Logstash 8.0 reaches GA, which is why I call out the value on the pre-release branch. I want people hitting this doc to have a "wait a minute" moment.

For the 7.x backport, the text will be more like:

> NOTE: Until the General Availability of Logstash 8.0.0 and the _final_ minor release of Logstash 7.x, any value for `pipeline.ecs_compatibility` other than `disabled` may have undesireable consequences when performing upgrades.
>      As we continue to release updated plugins with ECS-Compatibility modes, opting into them at a pipeline or process level will cause you to automatically consume breaking changes with each upgrade, which may change the shape of data your pipeline produces.

And 7._final_ will simply not need this caveat at all.